q2_diversity/tests/test_ordination.py

```diff
@@ -52,3 +53,37 @@ def test_pcoa_biplot(self):
 
         skbio.util.assert_ordination_results_equal(observed, expected,
                                                    ignore_directionality=True)
+
+
+class tsneTests(unittest.TestCase):
+
+    def setUp(self):
+        self.dm = skbio.DistanceMatrix([[0.0000000, 0.3333333, 0.6666667],
+                                        [0.3333333, 0.0000000, 0.4285714],
+                                        [0.6666667, 0.4285714, 0.0000000]],
+                                       ids=['S1', 'S2', 'S3'])
+
+    def test_tsne_default(self):
+        observed = tsne(self.dm).samples.shape
+        self.assertEqual(observed, (3, 3))
```

Comment:
Is there anything else we can test here, besides the dimensions? Is this test stochastic? If not, perhaps you can add an assertion that verifies the right values were calculated.

Reply:
Different runs with the same matrix can produce different results(for both t-SNE and UMAP). Variance between runs are relatively small but it is not possible to predict the exact results.